### PR TITLE
[dagster-dbt][refactor] Use AssetOut.from_spec() to create AssetOuts for dbt_assets decorator

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title -- prepares us for a world in which the DagsterDbtTranslator just returns an AssetSpec

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26613?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26614** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26614?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26613** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26613?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26613?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#23824** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23824?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
 from dagster._core.definitions.tags import build_kind_tag
 from dagster._utils.merger import merge_dicts
 
+from dagster_dbt.asset_specs import AssetSpec
```
Comment (maximearmstrong) [inline python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py:56]: Looks like this causes a circular import

```suggestion
from dagster import AssetSpec
```

**Review by maximearmstrong — APPROVED** [review]
Approved pending resolution of the circular import

**Comment by OwenKephart** [comment]
### Merge activity

* **Dec 20, 11:25 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/26613).

**Pull request merged**